cwms-radar-client/src/main/java/mil/army/usace/hec/cwms/radar/client/controllers/TextTimeSeriesController.java

```diff
@@ -76,4 +84,38 @@ public void deleteTimeSeries(ApiConnectionInfo apiConnectionInfo, TextTimeSeries
                 .execute()
                 .close();
     }
+
+    public String getTextValueFromUrl(ApiConnectionInfo apiConnectionInfo, RegularTextTimeSeriesRow row) {
+        String valueUrl = row.getValueUrl();
+        String downloadedText = null;
+        try {
+            downloadedText = download(apiConnectionInfo, valueUrl);
+        } catch(IOException e) {
+            logger.log(Level.FINE, "Failed to download data from " + valueUrl + "with exception " + e);
+        }
+        return downloadedText;
+    }
```

Comment:
Not every row will have a valueUrl.   Probably need null/isEmpty checks on valueUrl

Reply:
It should throw an exception if null